projects/opentrain/gtfs_pipeline/data_handler/DataHandler.java

```diff
@@ -1,24 +1,560 @@
+import org.apache.commons.csv.CSVRecord;
+import org.omg.CORBA.portable.InputStream;
+
+import java.io.FileInputStream;
+import java.io.InputStreamReader;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+
 import java.io.IOException;
+import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.nio.file.StandardOpenOption;
 import java.util.Arrays;
 import java.util.List;
 import java.lang.String;
+import java.lang.Iterable;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import java.util.Iterator;
+import java.lang.Object;
+import java.io.OutputStream;
+import java.io.FileOutputStream;
+import java.io.FileNotFoundException;
+import java.io.Reader;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileInputStream;
 
 class DataHandler {
-      public static void main(String[] args) throws IOException {
-        String output = "";
-        Path outputPath = Paths.get(args[args.length-1]);
-        for (String path : args) {
-            // TODO: Read the files in a streaming manner since they are large
-            Path curPath = Paths.get(path);
-            output += curPath.getFileName() + " lines: ";
-            Files.write(outputPath, new String(curPath.getFileName()+"\n").getBytes(), StandardOpenOption.APPEND, StandardOpenOption.CREATE);
-        }
-        System.out.print(output);
+
+  public static void saveAgency(Path csvPath, String path) {
+    try {
+      // where we serialize the messages
+      FileOutputStream output = new FileOutputStream(path);
+      // Creates a parser for the given path (csvPath argument)
+      CSVParser parser = CSVParser.parse(csvPath, StandardCharsets.UTF_8, CSVFormat.DEFAULT
+        .withHeader("agency_id", "agency_name", "agency_url", "agency_timezone", "agency_lang", "agency_phone", "agency_fare_url"));
+      // Creat iterator - to create from each record a message
+      Iterator<CSVRecord> iterator = parser.iterator();
+      // We need to ignore the first CSVRecord which holds the csv headers
+      iterator.next();
+      // messageCount will keep tracking how many message we serilize in the file.
+      int messageCount = 0;
+      while (iterator.hasNext()) {
+        CSVRecord record = iterator.next();
+        Agency.Builder agency = Agency.newBuilder();
+        agency.setAgencyId(Integer.parseInt(record.get("agency_id")))
+              .setAgencyName(record.get("agency_name"))
+              .setAgencyUrl(record.get("agency_url"))
+              .setAgencyTimezone(record.get("agency_timezone"))
+              .setAgencyLang(record.get("agency_lang"))
+              .setAgencyPhone(record.get("agency_phone"))
+              .setAgencyFareUrl(record.get("agency_fare_url"))
+              .build()
+              .writeDelimitedTo(output);
+        messageCount++;

```

Comment:
What actually do we do with this variable in the end? If it isn't used we can delete it. Also below.